internal/blocksync/pool.go

```diff
@@ -264,10 +269,15 @@ func (pool *BlockPool) RedoRequest(height int64) types.NodeID {
 
 // AddBlock validates that the block comes from the peer it was expected from and calls the requester to store it.
```

Comment:
Maybe mention which extended commit this requires, too?
In cases where we don't have an extended commit, can the caller pass `nil` safely?

Reply:
Agree